### PR TITLE
Update Lodash-related coding guidelines

### PR DESCRIPTION
**Pull request by sgomes**
#### Changes proposed in this Pull Request

* Update Lodash-related coding guidelines

#### Testing instructions

Documentation changes only; no need for testing.

### CODE REVIEW

**Review by wp-desktop — DISMISSED** [review]
WordPress Desktop CI Failure for job "wp-desktop-mac".

@sgomes please inspect this job's build steps for breaking changes at [this link](https://circleci.com/gh/Automattic/wp-calypso/947402). For temporal failures, you may try to "Rerun Workflow from Failed".

Please also ensure this branch is rebased off latest Calypso.

**Comment by sgomes** [comment]
Looks like the desktop tests are having some issues, if they're failing on a documentation-only PR 😕

File: docs/coding-guidelines/javascript.md [hunk 1]
```diff
@@ -559,8 +562,8 @@ prop = object[ 'default' ];
 prop = object[ 'key-with-hyphens' ];
 ```
 
-That said, avoid accessing nested properties through a chain of dot notation as this can lead cause access errors.
-Instead, use Lodash's [`get`](https://lodash.com/docs#get) function. It will safely handle cases where a property or object is missing at any point in the nesting chain.
+For nested properties, avoid using Lodash's [`get`](https://lodash.com/docs#get) function.
+Instead, you can use the standard optional chaining syntax (`?.`) after properties that may be missing. That will safely handle cases where a property or object is missing at any point in the nesting chain.
```
Comment (tyxla) [inline docs/coding-guidelines/javascript.md:566]: Should we also add a sentence to suggest nullish coalescing usage here?

**Review by tyxla — APPROVED** [review]
Looks great, thanks for writing it up 🙇‍♂️ 

I've left a couple of notes, feel free to ship with or without addressing them. 🚢

File: docs/coding-guidelines/javascript.md [hunk 2]
```diff
@@ -314,37 +314,37 @@ if ( someValue == null ) {
...
-- String: [`isString( value )`](https://lodash.com/docs#isString)
-- Number: [`isNumber( value )`](https://lodash.com/docs#isNumber)
-- Boolean: [`isBoolean( value )`](https://lodash.com/docs#isBoolean)
-- Object: [`isPlainObject( value )`](https://lodash.com/docs#isPlainObject)
-- null: [`isNull( value )`](https://lodash.com/docs#isNull)
-- undefined: [`isUndefined( value )`](https://lodash.com/docs#isUndefined)
-- undefined or null (either): [`isNil( value )`](https://lodash.com/docs#isNil)
+If you really must check the type of a value, however, do the following:
+
+- String: [`isString( value )`](https://lodash.com/docs#isString).
```
[the remark this answers is not in the document]
Comment (sgomes) [reply docs/coding-guidelines/javascript.md]: Changed, and added an explanation 👍

File: docs/coding-guidelines/javascript.md (hunk 1 above)
Comment (sgomes) [reply docs/coding-guidelines/javascript.md:566]: Good idea! Done.

**Comment by sgomes** [comment]
Thanks for the review, @tyxla! Addressed all your comments 👍

**Pull request merged**